platform/net/ubuntu_net_manager_test.go

```diff
@@ -904,6 +938,157 @@ iface ethstatic inet static
+				interfaceAddrsProvider.GetInterfaceAddresses = []boship.InterfaceAddress{
+					boship.NewSimpleInterfaceAddress("ethstatic", "1.2.3.5"),
+				}
+				portableNetwork = boshsettings.Network{
+					Type:     "manual",
+					IP:       "10.112.166.136",
+					Netmask:  "255.255.255.192",
+					Resolved: false,
+					UseDHCP:  false,
+					DNS:      []string{"8.8.8.8"},
+					Alias:    "eth0:0",
+				}
+				staticNetwork = boshsettings.Network{
+					Type:     "dynamic",
+					IP:       "169.50.68.75",
+					Netmask:  "255.255.255.224",
+					Gateway:  "169.50.68.65",
+					Default:  []string{"gateway", "dns"},
+					Resolved: false,
+					UseDHCP:  false,
+					DNS:      []string{"8.8.8.8", "10.0.80.11", "10.0.80.12"},
+					Mac:      "06:64:d4:7d:63:71",
+					Alias:    "eth1",
+				}
+				staticNetwork1 = boshsettings.Network{
+					Type:     "dynamic",
+					IP:       "10.112.39.113",
+					Netmask:  "255.255.255.128",
+					Resolved: false,
+					UseDHCP:  false,
+					DNS:      []string{"8.8.8.8", "10.0.80.11", "10.0.80.12"},
+					Mac:      "06:b7:e8:0c:38:d8",
+					Alias:    "eth0",
+				}
+				interfaceAddrsProvider.GetInterfaceAddresses = []boship.InterfaceAddress{
+					boship.NewSimpleInterfaceAddress("eth0", "10.112.39.113"),
+					boship.NewSimpleInterfaceAddress("eth1", "169.50.68.75"),
+				}
+				fs.WriteFileString("/etc/resolv.conf", `
+nameserver 8.8.8.8
+nameserver 10.0.80.11
+nameserver 10.0.80.12
+`)
+			})
+
+			It("succeeds", func() {
+				stubInterfaces(map[string]boshsettings.Network{
+					"eth1": staticNetwork,
+					"eth0": staticNetwork1,
+				})
+
+				errCh := make(chan error)
+				err := netManager.SetupNetworking(boshsettings.Networks{"default": portableNetwork, "dynamic": staticNetwork, "dynamic_1": staticNetwork1}, errCh)
+				Expect(err).NotTo(HaveOccurred())
+
+				broadcastErr := <-errCh // wait for all arpings
+				Expect(broadcastErr).ToNot(HaveOccurred())
+
+				Expect(addressBroadcaster.BroadcastMACAddressesAddresses).To(Equal([]boship.InterfaceAddress{
+					boship.NewSimpleInterfaceAddress("eth0", "10.112.39.113"),
+					boship.NewSimpleInterfaceAddress("eth1", "169.50.68.75"),
+				}))
+
+				networkConfig := fs.GetFileTestStat("/etc/network/interfaces")
+				Expect(networkConfig).ToNot(BeNil())
+
+				expectedNetworkConfigurationForStaticAndDhcp = `# Generated by bosh-agent
+auto lo
+iface lo inet loopback
+
+auto eth0
+iface eth0 inet static
+    address 10.112.39.113
+    network 10.112.39.0
+    netmask 255.255.255.128
+
+auto eth0:0
+iface eth0:0 inet static
+    address 10.112.166.136
+    network 10.112.166.128
+    netmask 255.255.255.192
+
+auto eth1
+iface eth1 inet static
+    address 169.50.68.75
+    network 169.50.68.64
+    netmask 255.255.255.224
+    broadcast 169.50.68.95
+    gateway 169.50.68.65
+
+dns-nameservers 8.8.8.8 10.0.80.11 10.0.80.12`
+				Expect(networkConfig.StringContents()).To(Equal(expectedNetworkConfigurationForStaticAndDhcp))
+			})
+
+			It("fails", func() {
```

Comment:
The test for the `fails` block is the same as the `succeeds` block above. Could you please update this test?